### PR TITLE
new tab onboarding

### PR DESCRIPTION
**Pull request by SabrinaTardio**
Task/Issue URL: https://app.asana.com/0/1204186595873227/1207569384066444/f

**Description**: Implements new tab page messages for the new onboarding. The new or old messages will shown depending on the variant manager

**Steps to test this PR**:
1. killall Simulator , run xcrun simctl erase all then launch from xcode again
2. You should see either the old or the new onboarding

Old onboarding
1. Go through the old onboarding
2. At the end of it the new tab page should show the “Next, try visiting one of your favorite sites!\n\nI’ll block trackers so they can’t spy on you. I’ll also upgrade the security of your connection if possible. 🔒” dialog
3. after performing a search or visiting a site it should show the “You’ve got this!\n\nRemember: Every time you browse with me, a creepy ad loses its wings. 👍” dialog
4. After it should always show a normal new tab page

New onboarding
1. Go through the initial new onboarding
2. At the end of the onboarding it should show the new search suggestion dialog
3. Either click or perform a search
4. Open a new tab page and this should invite you to visit a website (if you don’t visit a website and open a new tab page it should keep showing you this invitation
5. After visiting a site it should show you the final dialog. Clicking on high five should show the standard new tab page. 
6. After it should always show a normal new tab page

<!—
Before submitting a PR, please ensure you have tested the combinations you expect the reviewer to test, then delete configurations you *know* do not need explicit testing.

Using a simulator where a physical device is unavailable is acceptable.
—>

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `’`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

—
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/HomeViewController+DaxDialogs.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        daxDialogViewController.loadViewIfNeeded()
+        daxDialogViewController.message = spec.message
+        daxDialogViewController.accessibleMessage = spec.accessibilityLabel
+
+        view.addGestureRecognizer(daxDialogViewController.tapToCompleteGestureRecognizer)
+
+        daxDialogContainerHeight.constant = daxDialogViewController.calculateHeight()
+        hideLogo()
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```
Comment (brindy) [inline DuckDuckGo/HomeViewController+DaxDialogs.swift:43]: Where did this delay come from?  Was it always there?  It was surprising to me when I saw it, I would consider just making it smaller (less than 0.3 which is the threshold for being noticeable usually) if it's really needed or just get rid of it entirely.

File: DuckDuckGo/MainViewController.swift [hunk 2]
```diff
@@ -741,11 +744,16 @@ class MainViewController: UIViewController {
             addToContentContainer(controller: controller)
             viewCoordinator.logoContainer.isHidden = true
         } else {
-            let controller = HomeViewController.loadFromStoryboard(model: tabModel,
-                                                                   favoritesViewModel: favoritesViewModel,
-                                                                   appSettings: appSettings,
-                                                                   syncService: syncService,
-                                                                   syncDataProviders: syncDataProviders)
+            let newTabDaxDialogFactory = NewTabDaxDialogFactory(delegate: self)
+            let homePageDependencied = HomePageDependencies(model: tabModel,
```
Comment (brindy) [inline DuckDuckGo/MainViewController.swift]: typo here `homePageDependencied` -> `homePageDependencies`

File: DuckDuckGo/OnboardingExperiment/ContextualDaxDialogs/OnboardingSuggestionsViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    var itemsList: [ContextualOnboardingListItem] {
+        suggestedSitesProvider.list
+    }
+
+    func listItemPressed(_ item: ContextualOnboardingListItem) {
+        var urlString = item.title
+        var components = URLComponents(string: urlString)
+        if components?.scheme == nil {
+            components?.scheme = "https"
```
Comment (brindy) [inline DuckDuckGo/OnboardingExperiment/ContextualDaxDialogs/OnboardingSuggestionsViewModel.swift]: Can we just have all the items in the list have a scheme?

File: DuckDuckGoTests/ContextualOnboardingNewTabDialogFactoryTests.swift [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        let host = UIHostingController(rootView: view)
+        XCTAssertNotNil(host.view)
+
+        // Then
+        let addFavoriteDialog = find(ContextualDaxDialog.self, in: host)
+        XCTAssertNotNil(addFavoriteDialog)
+        XCTAssertEqual(addFavoriteDialog?.message.string, homeDialog.message)
+    }
+
+
```
Comment (brindy) [inline DuckDuckGoTests/ContextualOnboardingNewTabDialogFactoryTests.swift]: some extra space here

File: DuckDuckGoTests/ContextualOnboardingNewTabDialogFactoryTests.swift [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    private func find<T: View>(_ type: T.Type, in root: Any) -> T? {
```
Comment (brindy) [inline DuckDuckGoTests/ContextualOnboardingNewTabDialogFactoryTests.swift:107]: very nice, might be an idea to write up how to test swift view creation and share it with the team?

**Review by brindy — APPROVED** [review]
LGTM:

* Few minor comments for you to look at.
* Don't forget to bring this code up to date with `main` - I spotted something that looks a bit out of date (I removed something from the variant manager the other day but it's still there on this branch).

File: DuckDuckGo/HomeViewController+DaxDialogs.swift (hunk 1 above)
Comment (SabrinaTardio) [reply DuckDuckGo/HomeViewController+DaxDialogs.swift:43]: This is the old onboarding, I just moved the method but I have not touched anything about it.

**Pull request merged**